sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/BloomFilterMightContain.scala

```diff
@@ -59,12 +61,24 @@ case class BloomFilterMightContain(
             if !subquery.containsPattern(OUTER_REFERENCE) =>
             TypeCheckResult.TypeCheckSuccess
           case _ =>
-            TypeCheckResult.TypeCheckFailure(s"The Bloom filter binary input to $prettyName " +
-              "should be either a constant value or a scalar subquery expression")
+            DataTypeMismatch(
+              errorSubClass = "BLOOM_FILTER_BINARY_OP_WRONG_TYPE",
+              messageParameters = Map(
+                "functionName" -> toSQLId(prettyName),
+                "actual" -> toSQLType(bloomFilterExpression.dataType)
+              )
+            )
         }
-      case _ => TypeCheckResult.TypeCheckFailure(s"Input to function $prettyName should have " +
-        s"been ${BinaryType.simpleString} followed by a value with ${LongType.simpleString}, " +
-        s"but it's [${left.dataType.catalogString}, ${right.dataType.catalogString}].")
+      case _ =>
+        DataTypeMismatch(
+          errorSubClass = "BLOOM_FILTER_WRONG_TYPE",
+          messageParameters = Map(
+            "functionName" -> toSQLId(prettyName),
+            "required" -> (s"${toSQLType(BinaryType)} followed by " +
+              s"a value with ${toSQLType(LongType)}"),
+            "actual" -> s"[${toSQLType(left.dataType)}, ${toSQLType(right.dataType)}]"
```

Comment:
Leave only `toSQLType(left.dataType)`, and move `[]` to error-classes.json, please.